### PR TITLE
[Mellanox] Support SKU Mellanox-SN5600-C224O8

### PR DESCRIPTION
**Pull request by DavidZagury**
#### Why I did it
Support Mellanox-SN5600-C224O8

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Add relevant files to support the new SKU

#### How to verify it
Regression / QoS tests

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: device/mellanox/x86_64-nvidia_sn5600-r0/Mellanox-SN5600-C224O8/buffers_defaults_objects.j2 [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+        {% endif -%}
+        "ingress_lossless_pool": {
+            {% if dynamic_mode is not defined -%}
+            "size": "{{ ingress_lossless_pool_size }}",
+            "xoff": "{{ ingress_lossless_pool_xoff }}",
+            {% endif -%}
+            "type": "ingress",
+            "mode": "dynamic"
+        },
+        "egress_lossless_pool": {
```
Comment (r12f) [inline device/mellanox/x86_64-nvidia_sn5600-r0/Mellanox-SN5600-C224O8/buffers_defaults_objects.j2:35]: do we still need lossless pool? can we rename it to lossy?

Comment (r12f) [reply device/mellanox/x86_64-nvidia_sn5600-r0/Mellanox-SN5600-C224O8/buffers_defaults_objects.j2:35]: also looks like we still have 2 egress pool here, should we remove one?

**Review by r12f — APPROVED** [review]

**Comment by r12f** [comment]
@noaOrMlnx @DavidZagury do you mind to help move this PR out of draft state as discussed in the meeting, so we can be unblocked with testbed build up? thanks!

File: device/mellanox/x86_64-nvidia_sn5600-r0/Mellanox-SN5600-C224O8/buffers_defaults_objects.j2 (hunk 1 above)
Comment (r12f) [reply device/mellanox/x86_64-nvidia_sn5600-r0/Mellanox-SN5600-C224O8/buffers_defaults_objects.j2:35]: sync'ed offline. this should not be a blocker anymore.

File: device/mellanox/x86_64-nvidia_sn5600-r0/Mellanox-SN5600-C224O8/qos.json.j2 [hunk 2]
```diff
@@ -0,0 +1 @@
+../../x86_64-mlnx_msn2700-r0/ACS-MSN2700/qos.json.j2
```
[the remark this answers is not in the document]
Comment (r12f) [reply device/mellanox/x86_64-nvidia_sn5600-r0/Mellanox-SN5600-C224O8/qos.json.j2:1]: Hi @eddyk-nvidia , do you mind to help confirm this question? As we sync'ed last time, the queues and DWRR settings can be created directly using SAI APIs. The alpha value will be provided by the pg lookup ini, so we don't need anything extra in the HWSKU.

**Review by liat-grozovik — APPROVED** [review]

**Review by liat-grozovik — APPROVED** [review]

**Comment by noaOrMlnx** [comment]
@bingwang-ms can you please help to cherry-pick to 202405?

**Pull request merged**